api/src/utils/itis-sort.ts

```diff
@@ -0,0 +1,141 @@
+
+  // Prioritize taxa where either the scientific name or any common name is EXACTLY EQUAL to the search terms joined
+  // eg. ['Wolf'] -> "Wolf" is prioritized over "Forest Wolf"
+  const exactlyEquals = customSortEqualsSearchTermsExact(containsAnyMatchJoined, searchTermsLower);
+
+  return exactlyEquals;
+};
+
+/**
+ * Sorts the ITIS response to prioritize records where any word in the scientific or
+ * common name matches ANY of the search terms
+ *
+ * @param {ItisSolrSearchResponse[]} data
+ * @param {string[]} searchTerms
+ * @memberof ItisService
+ */
+export const customSortContainsAnyMatchingSearchTerm = (
+  data: TaxonSearchResult[],
+  searchTerms: string[]
+): TaxonSearchResult[] => {
+  // Custom sorting function
+  const customSort = (a: TaxonSearchResult, b: TaxonSearchResult) => {
+    const aInReference = checkForMatch(a, searchTerms);
+    const bInReference = checkForMatch(b, searchTerms);
+
+    if (aInReference && !bInReference) {
+      return -1; // Place items from searchTerms before other items
+    } else if (!aInReference && bInReference) {
+      return 1; // Place other items after items from searchTerms
+    } else {
+      return 0; // Maintain the original order if both are from searchTerms or both are not
+    }
+  };
+
+  const checkForMatch = (item: TaxonSearchResult, searchTerms: string[]) => {
+    const commonNameWords = item.commonNames?.flatMap((name: string) => name.toLowerCase().split(/\s+/));
+    const scientificNameWords = item.scientificName.toLowerCase().split(/\s+/);
+
+    // Check if any word in commonNames or scientificName matches any word in searchTerms
+    return searchTerms.some(
+      (searchTerm) => scientificNameWords?.includes(searchTerm) || commonNameWords?.includes(searchTerm)
+    );
+  };
+
+  return data.sort(customSort);
+};
+
+/**
+ * Sorts the ITIS response to prioritize records where either the scientific name or
+ * any common name CONTAINS the search terms joined
+ *
+ * @param {ItisSolrSearchResponse[]} data
+ * @param {string[]} searchTerms
+ * @memberof ItisService
+ */
+export const customSortContainsSearchTermsJoinedExact = (
+  data: TaxonSearchResult[],
+  searchTerms: string[]
+): TaxonSearchResult[] => {
+  const customSort = (a: TaxonSearchResult, b: TaxonSearchResult) => {
+    const aInReference = checkForMatch(a, searchTerms);
+    const bInReference = checkForMatch(b, searchTerms);
+
+    if (aInReference && !bInReference) {
+      return -1; // Place items from searchTerms before other items
+    } else if (!aInReference && bInReference) {
+      return 0; // Maintain the original order
+    } else {
+      return 0; // Maintain the original order
+    }
+  };
+
+  // Function to check if an item is a match with search terms
+  const checkForMatch = (item: TaxonSearchResult, searchTerms: string[]) => {
+    const commonNameWords = item.commonNames?.map((name: string) => name.toLowerCase());
+    const scientificNameWord = item.scientificName.toLowerCase();
```

Comment:
I might comb over these methods and consider renaming a few variables. For example, `scientificNameWord` does not describe what the variable contains any better than calling it `scientificName` would. However, if it was instead called something like `scientificNameSanitized`, I would understand that it contains the scientific name, but normalized in some way (in this instance, converted to lowercase).

Reply:
On the topic of converting the scientific name to lowercase, you could make the code [more dry](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) by sanitizing the scientfic names and search terms in the parent method, allowing you to drop all the instances of `.toLowerCase()`. You could also sanitize these terms in other ways, such as trimming whitespace with `.trim`.